controllers/opentelemetrycollector_controller.go

```diff
@@ -168,6 +169,10 @@ func (r *OpenTelemetryCollectorReconciler) Reconcile(ctx context.Context, req ct
 func (r *OpenTelemetryCollectorReconciler) RunTasks(ctx context.Context, params reconcile.Params) error {
 	for _, task := range r.tasks {
 		if err := task.Do(ctx, params); err != nil {
+			// If we get an error that occur because a pod is being terminated then exit this loop
+			if apierrors.IsForbidden(err) && strings.Contains(err.Error(), "because it is being terminated") {
```

Comment:
one last thing, should we print the termination logline before? And maybe an extra hint, that we cancel the reconciliation?

Reply:
@frzifus what log level should I print that at?

Reply:
i thought maybe:
```go
r.log.Error(err, fmt.Sprintf("failed to reconcile %s", task.Name))
```
And maybe an additional line that says that the reconciliation will be canceled?

Reply:
I am a bit skeptical about checking the string `because it is being terminated"`. This could easily change in the future

Reply:
You might be able to check for `v1.NamespaceTerminatingCause` in the error: https://github.com/kubernetes/apiserver/blob/v0.26.0-rc.0/pkg/admission/plugin/namespace/lifecycle/admission.go#L177

Reply:
@damemi Thanks!